src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java

```diff
@@ -200,42 +309,122 @@ private static void checkTransformer3W(Transformer3WInput transformer3W) {
-    checkMinimumTapPositionIsLowerThanMaximumTapPosition(transformer3WType);
-    checkNeutralTapPositionLiesBetweenMinAndMaxTapPosition(transformer3WType);
+  protected static List<Try<Void, InvalidEntityException>> checkTransformer3WType(
+      Transformer3WTypeInput transformer3WType) {
+    try {
+      checkNonNull(transformer3WType, "a three winding transformer type");
+    } catch (InvalidEntityException e) {
+      return List.of(
+          new Failure<>(
+              new InvalidEntityException(
+                  "Validation not possible because received object {"
+                      + transformer3WType
+                      + "} was null",
+                  e)));
+    }
+
+    return Try.ofVoid(
+        InvalidEntityException.class,
+        () ->
+            detectNegativeQuantities(
+                new Quantity<?>[] {transformer3WType.getgM(), transformer3WType.getdPhi()},
+                transformer3WType),
+        () ->
+            detectZeroOrNegativeQuantities(
+                new Quantity<?>[] {
+                  transformer3WType.getsRatedA(),
+                  transformer3WType.getsRatedB(),
+                  transformer3WType.getsRatedC(),
+                  transformer3WType.getvRatedA(),
+                  transformer3WType.getvRatedB(),
+                  transformer3WType.getvRatedC(),
+                  transformer3WType.getrScA(),
+                  transformer3WType.getrScB(),
+                  transformer3WType.getrScC(),
+                  transformer3WType.getxScA(),
+                  transformer3WType.getxScB(),
+                  transformer3WType.getxScC()
+                },
+                transformer3WType),
+        () ->
+            detectPositiveQuantities(
+                new Quantity<?>[] {transformer3WType.getbM()}, transformer3WType),
+        () -> checkVoltageMagnitudeChangePerTapPosition(transformer3WType),
+        () -> checkMinimumTapPositionIsLowerThanMaximumTapPosition(transformer3WType),
+        () -> checkNeutralTapPositionLiesBetweenMinAndMaxTapPosition(transformer3WType));
   }
 
   /**
    * Validates a switch if: <br>
    * - its connected nodes are in the same voltage level
    *
    * @param switchInput Switch to validate
+   * @return a try object either containing an {@link InvalidEntityException} or an empty Success
    */
-  private static void checkSwitch(SwitchInput switchInput) {
-    if (!switchInput.getNodeA().getVoltLvl().equals(switchInput.getNodeB().getVoltLvl()))
-      throw new InvalidEntityException("Switch connects two different voltage levels", switchInput);
+  private static Try<Void, InvalidEntityException> checkSwitch(SwitchInput switchInput) {
+    if (!switchInput.getNodeA().getVoltLvl().equals(switchInput.getNodeB().getVoltLvl())) {
+      return Failure.ofVoid(
+          new InvalidEntityException("Switch connects two different voltage levels", switchInput));
+    } else {
+      return Success.empty();
```

Comment:
Maybe for these kind of pieces of code (there are many), a 
```java
Try.ofVoid(boolean success, VoidSupplier<E> supplier, Class<E> clazz)
```
would be advantageous.